### PR TITLE
Manage social media information in redux

### LINKED ISSUE #254: Tie together social media components
- Twitter message text
- Facebook message text
- Facebook preview
- Twitter preview

This can all be managed in Redux.

### PR DESCRIPTION
**Pull request by wbprice**
rather than a specific component.

Fixes #254

### CODE REVIEW

File: api/services/SurveyResultService.js [hunk 1]
```diff
@@ -146,14 +147,14 @@ module.exports = class SurveyResultService extends Service {
                 return a;
             }, []);
 
-            return matches.map(m => {
+            return orderBy(matches.map(m => {
```
Comment (ttavenner) [inline api/services/SurveyResultService.js]: @wbprice What is the reason behind moving the sort to the service?

File: frontend/js/components/organisms/SocialMediaIcons.js [hunk 2]
```diff
@@ -56,10 +44,8 @@ class SocialMediaSharing extends Component {
...
 SocialMediaSharing.propTypes = {
-    matches: PropTypes.array,
-    primary: PropTypes.bool,
-    name: PropTypes.string,
-    passPhrase: PropTypes.string
+    social: PropTypes.object,
+    primary: PropTypes.bool
 };
 
 export default SocialMediaSharing;
```
Comment (ttavenner) [inline frontend/js/components/organisms/SocialMediaIcons.js]: I receive an error when running `npm test` that I don't know how to resolve.

```
okcandidate-platform/dist/components/organisms/SocialMediaIcons.js:123
exports.default = (0, _reactRouter2.default)(SocialMediaSharing);
                                            ^

TypeError: (0 , _reactRouter2.default) is not a function
    at Object.<anonymous> (okcandidate-platform/dist/components/organisms/SocialMediaIcons.js:123:45)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (okcandidate-platform/dist/components/ecosystems/CandidateMatch.js:21:25)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (okcandidate-platform/dist/components/environments/Results.js:19:23)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (okcandidate-platform/dist/routes.js:77:16)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (okcandidate-platform/api/controllers/utils/getInitialRender.js:13:16)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (okcandidate-platform/api/controllers/ViewController.js:9:5)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (okcandidate-platform/api/controllers/index.js:4:26)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (okcandidate-platform/api/index.js:3:23)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (okcandidate-platform/index.js:5:15)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (okcandidate-platform/test/integration/controllers/AuthController.test.js:5:13)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at okcandidate-platform/node_modules/mocha/lib/mocha.js:220:27
    at Array.forEach (native)
    at Mocha.loadFiles (okcandidate-platform/node_modules/mocha/lib/mocha.js:217:14)
    at Mocha.run (okcandidate-platform/node_modules/mocha/lib/mocha.js:469:10)
    at Object.<anonymous> (okcandidate-platform/node_modules/mocha/bin/_mocha:404:18)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.runMain (module.js:604:10)
    at run (bootstrap_node.js:393:7)
    at startup (bootstrap_node.js:150:9)
    at bootstrap_node.js:508:3
```

File: api/services/SurveyResultService.js (hunk 1 above)
Comment (ttavenner) [inline api/services/SurveyResultService.js]: What is the reason behind moving the sort to the service?

**Review by ttavenner — CHANGES_REQUESTED** [review]

File: api/services/SurveyResultService.js (hunk 1 above)
Comment (wbprice) [reply api/services/SurveyResultService.js]: I needed to move sorting out of the results component to get the right info for social sharing, but the sorting could also happen in result-actions or result-reducer

Comment (ttavenner) [reply api/services/SurveyResultService.js]: That would be better. It is cleaner design not to assume any ordering of the data returned by the API.

File: frontend/js/redux/reducers/social-reducer.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+function getDescription(matches) {
+    return `${getShortDescription(matches)} ${initialState.description}`;
+}
+
+function getShortDescription(matches) {
+    return `I matched with ${candidateMatchString(matches)}.`;
+}
+
+function candidateMatchString(matches) {
```
Comment (qwo) [inline frontend/js/redux/reducers/social-reducer.js:26]: you can simplify and make this function more generic by doing the following 😄 
```
return matches.slice(0,-1).join(', ') + `, and ${matches.slice(-1)[0]}`
```

**Review by ttavenner — APPROVED** [review]

**Pull request merged**